Sources/Core/StringExtensions.swift

```diff
@@ -52,31 +52,114 @@ prefix func --> (body: () -> [String]) -> String {
     return -->body()
 }
 
+// Most of these are derived from https://www.binpress.com/tutorial/objective-c-reserved-keywords/43
+// other language conflicts should be ideally added here.
+// TODO: Find a way to separate this by language since the reserved keywords will differ.
 let objectiveCReservedWordReplacements = [
     "description": "description_text",
     "id": "identifier"
-  // TODO: Fill out more objc keywords with replacements.
 ]
 
+let objectiveCReservedWords = Set<String>([
+    "@catch()",
+    "@class",
+    "@dynamic",
+    "@end",
+    "@finally",
+    "@implementation",
+    "@interface",
+    "@private",
+    "@property",
+    "@protected",
+    "@protocol",
+    "@public",
+    "@selector",
+    "@synthesize",
+    "@throw",
+    "@try",
+    "BOOL",
+    "Class",
+    "IMP",
+    "NO",
+    "NULL",
+    "Protocol",
+    "SEL",
+    "YES",
+    "_Bool",
+    "_Complex",
+    "_Imaginery",
+    "atomic",
+    "auto",
+    "break",
+    "bycopy",
+    "byref",
+    "case",
+    "char",
+    "const",
+    "continue",
+    "default",
+    "do",
+    "double",
+    "else",
+    "enum",
+    "extern",
+    "float",
+    "for",
+    "goto",
+    "id",
+    "if",
+    "in",
+    "inline",
+    "inout",
+    "int",
+    "long",
+    "nil",
+    "nonatomic",
+    "oneway",
+    "out",
+    "register",
+    "restrict",
+    "retain",
+    "return",
+    "self",
+    "short",
+    "signed",
+    "sizeof",
+    "static",
+    "struct",
+    "super",
+    "switch",
+    "typedef",
+    "union",
+    "unsigned",
+    "void",
+    "volatile",
+    "while"
+])
+
 extension String {
     /// All components separated by _ will be capitalized including the first one
     func snakeCaseToCamelCase() -> String {
         var str: String = self
 
-        if let replacementString = objectiveCReservedWordReplacements[self] as String? {
+        if let replacementString = objectiveCReservedWordReplacements[self.lowercased()] as String? {
             str = replacementString
         }
 
         let components = str.components(separatedBy: "_")
         let name = components.map { return $0.uppercaseFirst }
-        return name.joined(separator: "")
+        let formattedName = name.joined(separator: "")
+        if objectiveCReservedWords.contains(formattedName) {
```

Comment:
Can this clash if you would like to use "void" for something but it's a reserved word in objective c?

Reply:
if you used "void" it would just become "voidProperty" or "VoidProperty"

Reply:
The purpose of this PR is to prevent reserved words from being generated as identifiers. I don't believe this will cause issues for most since naming anything as a reserved keyword is probably unlikely.